src/RecoveryServices/RecoveryServices.Backup.Providers/AzureWorkloadProviderHelper.cs

```diff
@@ -372,20 +372,101 @@ public AzureWorkloadProviderHelper(ServiceClientAdapter serviceClientAdapter)
+                {
+                    throw new ArgumentException(String.Format(Resources.InvalidScheduleWindowDuration, PolicyConstants.AfsHourlyWindowDurationMin, PolicyConstants.AfsHourlyWindowDurationMax));
+                }
+
+                DateTime windowStartTime = (DateTime)policy.ScheduleWindowStartTime;
+                DateTime minimumStartTime = new DateTime(windowStartTime.Year, windowStartTime.Month, windowStartTime.Day, 00, 00, 00, 00, DateTimeKind.Utc);
+                DateTime maximumStartTime = new DateTime(windowStartTime.Year, windowStartTime.Month, windowStartTime.Day, 19, 30, 00, 00, DateTimeKind.Utc);
+
+                //validate window start time 
+                if (windowStartTime > maximumStartTime || windowStartTime < minimumStartTime)
+                {
+                    throw new ArgumentException(String.Format(Resources.ScheduleWindowStartTimeOutOfRange));
+                }
+
+                // final backup time can be 23:30:00                
+                DateTime finalBackupTime = new DateTime(windowStartTime.Year, windowStartTime.Month, windowStartTime.Day, 23, 30, 00, 00, DateTimeKind.Utc);
+                TimeSpan diff = finalBackupTime - windowStartTime;
+
+                // If ScheduleWindowDuration is greator than (23:30 - ScheduleWindowStartTime) then throw exception  
+                if (diff.TotalHours < policy.ScheduleWindowDuration)
+                {
+                    throw new ArgumentException(String.Format(Resources.InvalidLastBackupTime));
+                }
+            }
+        }
+
+        public void ValidateAzureIaasVMSchedulePolicy(CmdletModel.SchedulePolicyBase policy)
+        {            
+            if(policy.GetType() == typeof(CmdletModel.SimpleSchedulePolicy))
+            {                
+                CmdletModel.SimpleSchedulePolicy simpleSchedulePolicy = (CmdletModel.SimpleSchedulePolicy)policy;
+                
+                // Standard hourly is restricted for IaasVM
+                if (simpleSchedulePolicy.ScheduleRunFrequency == ScheduleRunType.Hourly)
+                {
+                    throw new ArgumentException(Resources.StandardHourlyPolicyNotSupported);
+                }
+            }  
+            else if (policy.GetType() == typeof(CmdletModel.SimpleSchedulePolicyV2))
+            {                   
+                CmdletModel.SimpleSchedulePolicyV2 simpleSchedulePolicyV2 = (CmdletModel.SimpleSchedulePolicyV2)policy;
+                if (simpleSchedulePolicyV2.ScheduleRunFrequency == ScheduleRunType.Hourly)
+                {                    
+                    throw new ArgumentException("Enhanced Hourly policy is currently not supported for WorkloadType AzureIaasVM. This will be supported soon");
+                                        
+                    /*List<int> AllowedScheduleIntervals = new List<int> { 4, 6, 8, 12 };
```

Comment:
Please remove uncommented part

Reply:
done.

Reply:
> Please remove uncommented part

Basically all of this code will be needed in next release onwards when we add MBPD support for IaasVM Enhanced policy